From 3f2d5bdcf2dabb0394d348f9a1b6b8a678428428 Mon Sep 17 00:00:00 2001
From: Alexander Morozov <gelu.speculum@gmail.com>
Date: Fri, 8 Apr 2022 13:44:59 +0300
Subject: [PATCH] Fix destroy method in allocator interface

It should be fully ready to be used without any modifications, hence -
templated.
---
 include/allocator.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/allocator.h b/include/allocator.h
index 614a8e1..f329bba 100644
--- a/include/allocator.h
+++ b/include/allocator.h
@@ -19,8 +19,10 @@ public:
         return new (ptr) T(std::forward<Args>(args)...);
     }
 
-    void destroy(const void * ptr)
+    template <class T>
+    void destroy(void * ptr)
     {
+        static_cast<T *>(ptr)->~T();
         deallocate(ptr);
     }
 };
-- 
2.24.3 (Apple Git-128)

